<launch>
   <!-- <arg name="cloud_in" default="/zed2i/zed_node/point_cloud/cloud_registered"/> -->
    <!-- /carla/ego_vehicle/lidar -->
    <arg name="cloud_in" default="/carla/ego_vehicle/lidar"/>

    <arg name="target_frame" default="ego_vehicle" />
    <!-- <arg name="target_frame" default="zed2i_base_link" /> -->

    <arg name="laser_out_topic" default="zed/laser_scan"/>
    
    <node name='pointcloud_to_laserscan_node' pkg='pointcloud_to_laserscan' type='pointcloud_to_laserscan_node'>
      <remap from="cloud_in" to="$(arg cloud_in)"/>
      <remap from="scan" to="$(arg laser_out_topic)"/>
      <param name='target_frame' value="$(arg target_frame)"/>
      <rosparam>
         min_height  : 0.2
         max_height  : 1
         angle_min   : -1.570
         angle_max   : 1.5708
         
         angle_increment : 0.0087
         scan_time : 0.3333
         range_min : 0.45
         range_max : 100
         concurrency_level : 1
         use_inf : false
      </rosparam>
    </node>

</launch>